### PR TITLE
update list deps

### PR DESCRIPTION
**Pull request by amyXia1994**
## Feature description
1. support build deps graph and list upstream/downstream dependencies.  These apis do not reply on analyzing the Kusion stack/project structure, but only parse dependency by parsing import statements.
2. add apis: 
 - ListUpStreamFiles(workDir string, opt *DepOption) (deps []string, err error): ListUpStreamFiles return a list of upstream depend files from the given path list.
 - ListDownStreamFiles(workDir string, opt *DepOption) ([]string, error): ListDownStreamFiles return a list of downstream depend files from the given changed path list.

## Examples of usage:
- [the demo KCL](https://github.com/KusionStack/kclvm-go/tree/dev/xiarui.xr/update-list-deps/pkg/tools/list/testdata/complicate)
- the API call to list the upstream files that the focused files depend on:
`depParser, err := NewImportDepParser("testdata/complicate/", DepOption{Files: []string{appops/projectA/base/base.k", "appops/projectA/dev/main.k", "base/render/server/server_render.k"})
upstreamDeps := depParser.ListUpstreamFiles()`

### CODE REVIEW

File: kclvm.go [hunk 1]
```diff
@@ -127,9 +127,13 @@ func FormatPath(path string) (changedPaths []string, err error) {
 	return format.FormatPath(path)
 }
 
-// ListDepFiles return the depend files from the given path
-func ListDepFiles(workDir string, opt *ListDepFilesOption) (files []string, err error) {
-	return list.ListDepFiles(workDir, opt)
+// ListDepFiles return a graph that describes the depend files from the given path list
```
Comment (Peefy) [inline kclvm.go]: The return value is `deps`, and the comment is written to return the `graph`. Need to be unified?

File: kclvm_test.go [hunk 2]
```diff
@@ -263,7 +263,10 @@ func TestListDepFiles(t *testing.T) {
 		"main.k",
 		"app0/before/base.k",
 		"app0/main.k",
+		"app0/sub",
 		"app0/sub/sub.k",
+		"kcl_plugin/cmdb",
```
Comment (Peefy) [inline kclvm_test.go]: cmdb do not in the kcl plugin open source.

File: cmds/kcl-go/command/cmd_main.go [hunk 3]
```diff
@@ -71,7 +71,6 @@ func Main() {
 		NewAstCmd(),
 
 		NewListCmd(),
-		NewLispAppCmd(),
```
Comment (chai2010) [inline cmds/kcl-go/command/cmd_main.go]: `kcl-go list-app` is used in rest-api/ListDepFiles

File: kclvm.go [hunk 4]
```diff
@@ -127,9 +127,13 @@ func FormatPath(path string) (changedPaths []string, err error) {
...
+func ListDepFiles(workDir string, files []string, opt *ListDepFilesOption) (deps []string, err error) {
```
Comment (chai2010) [inline kclvm.go]: api changed, move the `files` to opt?

File: kclvm.go [hunk 5]
```diff
@@ -127,9 +127,13 @@ func FormatPath(path string) (changedPaths []string, err error) {
...
+	return list.ListDepFiles(workDir, files, opt)
+}
+
+func GetAffected(workDir string, files []string, opt *ListDepFilesOption, changedPaths []string) ([]string, error) {
```
Comment (chai2010) [inline kclvm.go]: `func ListChangedFiles(workDir string,  changedPaths []string, opt *ListDepFilesOption) ([]string, error)`

File: kclvm.go (hunk 1 above)
Comment (amyXia1994) [reply kclvm.go]: fixed

File: cmds/kcl-go/command/cmd_main.go (hunk 3 above)
Comment (amyXia1994) [reply cmds/kcl-go/command/cmd_main.go]: reverted

File: kclvm.go (hunk 4 above)
Comment (amyXia1994) [reply kclvm.go]: fixed

File: kclvm.go (hunk 5 above)
Comment (amyXia1994) [reply kclvm.go]: to keep the old ListDepFiles API, the interfaces to add renamed to:
ListUpStreamFiles and ListDownStreamFiles

File: kclvm_test.go (hunk 2 above)
Comment (amyXia1994) [reply kclvm_test.go]: removed

File: pkg/tools/list/list.go [hunk 6]
```diff
@@ -24,3 +24,33 @@ func ListDepFiles(workDir string, opt *Option) (files []string, err error) {
 
 	return files, nil
 }
+
+func ListUpStreamFiles(workDir string, opt *DepOption) (deps []string, err error) {
```
Comment (Peefy) [inline pkg/tools/list/list.go:31]: API functions are best commented

File: pkg/tools/list/list.go [hunk 7]
```diff
@@ -24,3 +24,33 @@ func ListDepFiles(workDir string, opt *Option) (files []string, err error) {
...
+		return nil, err
+	}
+	depParser, err := NewImportDepParser(pkgroot, *opt)
+	if err != nil {
+		return nil, err
+	}
+	return depParser.ListUpstreamFiles(), nil
+}
+
+func ListDownStreamFiles(workDir string, opt *DepOption) ([]string, error) {
```
Comment (Peefy) [inline pkg/tools/list/list.go:47]: API functions are best commented

File: pkg/spec/gpyrpc/gpyrpc.proto [hunk 8]
```diff
@@ -263,8 +266,8 @@ message ValidateCode_Result {
 }
 
 message CodeSnippet {
-    string schema = 1;
-    string rule = 2;
+	string schema = 1;
+	string rule = 2;
```
Comment (Peefy) [inline pkg/spec/gpyrpc/gpyrpc.proto:269]: Avoid unnecessary diffs.

File: pkg/spec/gpyrpc/gpyrpc.proto [hunk 9]
```diff
@@ -153,7 +155,8 @@ message ExecProgram_Args {
 
 	repeated string k_filename_list = 2;
 	repeated string k_code_list = 3;
-	
+
```
Comment (Peefy) [inline pkg/spec/gpyrpc/gpyrpc.proto:158]: Avoid unnecessary diffs.

**Review by chai2010 — DISMISSED** [review]
LGTM

File: pkg/tools/list/list.go (hunk 6 above)
Comment (amyXia1994) [reply pkg/tools/list/list.go:31]: fixed

File: pkg/tools/list/list.go (hunk 7 above)
Comment (amyXia1994) [reply pkg/tools/list/list.go:47]: fixed

File: pkg/spec/gpyrpc/gpyrpc.proto (hunk 8 above)
Comment (amyXia1994) [reply pkg/spec/gpyrpc/gpyrpc.proto:269]: these diffs are intended to keep consistent code format

File: pkg/spec/gpyrpc/gpyrpc.proto (hunk 9 above)
Comment (amyXia1994) [reply pkg/spec/gpyrpc/gpyrpc.proto:158]: these diffs are intended to keep consistent code format

**Review by Peefy — DISMISSED** [review]
LGTM.

**Comment by Peefy** [comment]
LGTM

**Review by Peefy — APPROVED** [review]
LGTM

**Review by chai2010 — APPROVED** [review]

**Pull request merged**